notebooks: document new auto-closure feature (#165)
reanahub/docs.reana.io: docs/administration/configuration/configuring-interactive-sessions/index.md

# Configuring interactive sessions

## Inactive interactive sessions cleanup

### Automatic closure

By default, interactive sessions in REANA are kept open indefinitely. However, administrators
have the flexibility to configure the maximum inactivity period after which interactive
sessions will be automatically closed. This helps optimize resource utilization and ensure that unused sessions
are not consuming unnecessary resources.

The automatic closure of interactive sessions in REANA can be customized by configuring
[Helm values](https://github.com/reanahub/reana/tree/master/helm/reana):

- `interactive_sessions.maximum_inactivity_period`:

>   This option allows you to set a limit (in days) for the maximum inactivity period of interactive sessions after which they will be closed.

Reviewer: I think we should also add show setting this value to `forever` disables the automatic closure of idle sessions

# Configuring interactive sessions

## Inactive interactive sessions cleanup

### Automatic closure

By default, interactive sessions in REANA are kept open indefinitely. However, administrators
have the flexibility to configure the maximum inactivity period after which interactive
sessions will be automatically closed. This helps optimize resource utilization and ensure that unused sessions
are not consuming unnecessary resources.

The automatic closure of interactive sessions in REANA can be customized by configuring
[Helm values](https://github.com/reanahub/reana/tree/master/helm/reana):

- `interactive_sessions.maximum_inactivity_period`:
  This option allows you to set a limit (in days) for the maximum inactivity period of interactive sessions after which they will be closed.

>   If this option is set to `forever`, interactive sessions will be kept open indefinitely unless manually closed.

Reviewer 2: ... will be kept open indefinitely until they are closed manually.
(This the default value.)

- `interactive_sessions.cronjob_schedule`:
  This option allows you to define how often the cleanup process should be scheduled, using a [cron expression](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#schedule-syntax).

For example, this is how you would configure the Helm chart to make REANA automatically close
interactive sessions that have been inactive for more than 7 days, and perform the cleanup

> check every day at 3 AM:

Reviewer 2: ... check every day at 03:00

(or "... at 3am")

```{.yaml .copy-to-clipboard}
interactive_sessions:
    maximum_inactivity_period: 7
    cronjob_schedule: "0 3 * * *"
```

> ### Manual closure

Reviewer 2: This section could be removed, since this is just a tip for admins.

We could actually transform it into a `!!! note` so that it would stand out after the example. The text could be simplified to read:

- Note that as an administrator you can also force the manual closure of opened interactive sessions regardless of the value of `interactive_sessions.maximum_inactivity_period` by means of the `reana-admin` tool. For example, to close any interactive session that has been inactive since 30 days, you would run: ...

We could use several such `kubectl` tips throughout the admin-related documentation.

Reviewer 2: Yes, using "tip" sounds nice, it would suit well these "howto" notes, and would differentiate them nicely from the "information" notes...

# Configuring interactive sessions

> ## Interactive sessions cleanup

Reviewer: What do you think about dropping this intermediate section?

Author: I think it would be useful especially if in the future we add more sections, unrelated to the closure of interactive sessions.
I reworded it:
```
## Inactive interactive sessions cleanup
   ### Automatic closure
   ### Manual closure
```

In my opinion it gives more structure to the documentation page and to the table of contents - do you agree?

> ### Automatic and scheduled closure

Reviewer: ```suggestion
### Automatic closure of inactive sessions
```
WDYT?

By default, interactive sessions in REANA are kept open indefinitely. However, administrators

> have the flexibility to configure the maximum inactivity period after which interactive
> sessions will be automatically closed; the schedule of this cleanup process can be
> customized. This helps optimize resource utilization and ensure that unused sessions

Reviewer: ```suggestion
can configure a maximum inactivity period after which interactive
sessions will be automatically closed. This helps optimize resource utilization and ensure that unused sessions
```

WYDT? I don't think it's necessary to already talk about schedule here, it  can be introduced later when we show the Helm values.

are not consuming unnecessary resources.

> The automatic closure of interactive sessions in REANA can be customized by configuring
> [Helm values](https://github.com/reanahub/reana/tree/master/helm/reana).
> The following Helm values allow you to control the behavior of the automatic closure feature:

Reviewer: ```suggestion
The automatic closure of inactive interactive sessions in REANA can be customized by means of the following
[Helm values](https://github.com/reanahub/reana/tree/master/helm/reana):
```

To avoid repetition of _automatic closure_

- `interactive_sessions.maximum_inactivity_period`:
  This option allows you to set a limit (in days) for the maximum inactivity period of interactive sessions after which they will be closed.

- `interactive_sessions.cronjob_schedule`:

>   This option allows you to set the schedule for the cleanup process using the cronjob format.

Reviewer: ```suggestion
  This option allows you to define how often the cleanup process should be scheduled, using a [cron expression](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#schedule-syntax).
```

Reviewer: On another note, could you also update the URL to the k8s docs in `configuring-user-quotas`, as it has changed in the meantime?

Example:

```yaml
interactive_sessions:
    maximum_inactivity_period: 7
    cronjob_schedule: "0 3 * * *"

> ...

Reviewer: What do you think about removing `...` and adding the copy-to-clipboard button to this snippet? Example: https://github.com/reanahub/docs.reana.io/blob/4149b8d8fa4b219a0ed4fdf74d5e4719fbe8b061/docs/administration/configuration/configuring-storage-volumes/index.md?plain=1#L40

Reviewer: The copy-to-clipboard button is not being shown, probably because there is a missing `.` before `yaml`

```

In the above example, interactive sessions that have been inactive for more than 7 days

> will be automatically closed, and the cleanup check will be performed every day at 3 AM.

Reviewer: What do you think about moving this before the YAML snippet, in order to avoid having _Example:_?

You could say something like _For example, this is how you would configure ..._

> ### Manual cleanup

Reviewer: ```suggestion
### Manual cleanup of inactive sessions
```

> In addition to the automatic closure and scheduled cleanup, administrators can also

Reviewer: ```suggestion
In addition to the periodic automatic closure, administrators can also
```

manually clean up interactive sessions that have been inactive for a certain number of days.

> 
> To manually clean up interactive sessions, use the `interactive-session-cleanup` command
> of the `reana-admin` tool that is available from the `reana-server` pod.

Reviewer: ```suggestion
This can be achieved with the `interactive-session-cleanup` command
of the `reana-admin` tool.
```

What do you think about moving this to the previous paragraph, while also making it shorter?

For example, to delete the interactive sessions that have been inactive for the past 30
days or more, you can use:

```console
$ kubectl exec -i -t deployment/reana-server -- flask reana-admin interactive-session-cleanup --days 30
Interactive session 'reana-run-session-69d590c3-ce47-4ae1-8719-bf8952627b37-c7djf67l' has been closed.
Interactive session 'reana-run-session-69d590c3-ce47-4ae1-8719-bf8952627b37-c7djf67l' was updated 2 days ago. Leaving opened.
```

This manual cleanup process provides administrators with additional control to manage

> inactive interactive sessions effectively.

Reviewer: I think we can remove this, wdyt?

Author: I removed the unnecessary last paragraph (_"This manual..."_), but I think it would be nice to show a small example that can be copied and pasted directly
